### PR TITLE
feat(OAuth (22 > main)): Google OAuth 2.0

### LINKED ISSUE #22: Google OAuth 2.0
Add the feature login with google, that implies:
- [ ] a button login with google (with a picture of google)
- [ ] use a token to login via google
here is a link that explain how to do it https://youtu.be/rRn2EisxPl4

### PR DESCRIPTION
**Pull request by Pablodeibar**
Closes #22 
actually created a component GoogleButton on Web that handle connection with google and create a user on DB it looks like this
image
return 403 if user already exist and 201 if created

### CODE REVIEW

File: server/src/auth/auth.controller.ts [hunk 1]
```diff
@@ -39,4 +39,15 @@ export class AuthController {
   signIn(@Body() params: AuthDto) {
     return this.authService.signIn(params);
   }
+
+  @Post('/loginService')
+  async loginService(@Body('token') token): Promise<any> {
```
Comment (Benjicatch) [inline server/src/auth/auth.controller.ts:44]: define type of token

**Review by Benjicatch — CHANGES_REQUESTED** [review]

File: server/src/auth/auth.service.ts [hunk 2]
```diff
@@ -67,4 +78,58 @@ export class AuthService {
...
+            id: newUser.id,
+            email: newUser.email,
+            randomToken: newUser.randomToken,
+            createdAt: newUser.createdAt,
+            updatedAt: newUser.updatedAt,
+          },
+        };
+      }
+
+      throw new ConflictException('User already exists');
```
[the remark this answers is not in the document]
Comment (Pablodeibar) [reply server/src/auth/auth.service.ts]: fixed

File: web/src/Component/GoogleButton/GoogleButton.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  return (
+    <div>
+        <GoogleOAuthProvider clientId={process.env.REACT_APP_GOOGLE_CLIENT_ID}>
+            <GoogleLogin
+
+              useOneTap
+
+              onSuccess={async credentialResponse => {
+                console.log(credentialResponse.credential);
```
[the remark this answers is not in the document]
Comment (Pablodeibar) [reply web/src/Component/GoogleButton/GoogleButton.tsx]: fixed

File: web/src/index.js [hunk 4]
```diff
@@ -2,12 +2,14 @@ import React from 'react';
...
 import './index.css';
 import App from './App';
+import Login from './Page/Login/login.tsx';
 import reportWebVitals from './reportWebVitals';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
   <React.StrictMode>
-    <App />
+    {/* <App /> */}
```
[the remark this answers is not in the document]
Comment (Pablodeibar) [reply web/src/index.js]: fixed

File: server/src/auth/auth.controller.ts (hunk 1 above)
Comment (Pablodeibar) [reply server/src/auth/auth.controller.ts]: fixed

File: server/src/auth/auth.service.ts [hunk 5]
```diff
@@ -67,4 +74,66 @@ export class AuthService {
...
+            id: newUser.id,
+            email: newUser.email,
+            randomToken: newUser.randomToken,
+            createdAt: newUser.createdAt,
+            updatedAt: newUser.updatedAt,
+          },
+        };
+      }
+
+      return {
```
Comment (Benjicatch) [inline server/src/auth/auth.service.ts:126]: before return generate a new random token for the user like lign 61

**Review by Benjicatch — CHANGES_REQUESTED** [review]

File: server/src/auth/auth.service.ts [hunk 6]
```diff
@@ -67,4 +78,58 @@ export class AuthService {
...
+        throw new ForbiddenException('Invalid Google token');
+      }
+
+      const existingUser = await this.prisma.user.findUnique({
+        where: {
+          email,
+        },
+      });
+
+      if (!existingUser) {
```
[the remark this answers is not in the document]
Comment (Benjicatch) [reply server/src/auth/auth.service.ts:98]:
```suggestion

      const service = await this.prisma.services.create({
        data: {
          userId: existingUser.id,
          token: // token of google
          typeService: ServiceType.GOOGLE,
        }
      });
```

File: server/src/auth/auth.service.ts (hunk 5 above)
Comment (Pablodeibar) [reply server/src/auth/auth.service.ts:156]: filled the service with bad infos caus i will do it on another branch caus it involve some change on my google cloud etc so another functionnality so on another branch

File: server/src/auth/auth.service.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (Pablodeibar) [reply server/src/auth/auth.service.ts:99]: filled the service with bad infos caus i will do it on another branch caus it involve some change on my google cloud etc so another functionnality so on another branch

**Review by Benjicatch — APPROVED** [review]
ok

**Review by AlexGuillard — APPROVED** [review]
ok

**Pull request merged**